From 06faac90d17950ddc7f2cd07a18bf6557bd5f0d3 Mon Sep 17 00:00:00 2001
From: Fan Zhicheng <r32736@freescale.com>
Date: Tue, 28 Dec 2010 14:39:31 +0800
Subject: [PATCH 572/670] ENGR00137224 L2Switch: Fix L2Switch port learn not
 take effect issue.

Fix Mx28 L2Switch port learning not works issue.

Signed-off-by: Fan Zhicheng <r32736@freescale.com>
---
 drivers/net/fec_switch.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/fec_switch.c b/drivers/net/fec_switch.c
index dc3f205..1ce0202 100644
--- a/drivers/net/fec_switch.c
+++ b/drivers/net/fec_switch.c
@@ -875,7 +875,7 @@ static void esw_get_port_learning(
 	struct switch_t  *fecp;
 
 	fecp = fep->hwp;
-	*ulPortLearning = (readl(&fecp->ESW_BKLR) & 0xff00) >> 16;
+	*ulPortLearning = (readl(&fecp->ESW_BKLR) & 0xff0000) >> 16;
 #ifdef DEBUG_PORT_LEARNING
 	printk(KERN_INFO "%s  fecp->ESW_BKLR %#lx\n",
 		__func__, readl(&fecp->ESW_BKLR));
-- 
1.8.0

